corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -286,6 +285,13 @@ def _diff_ledgers(self, case_ids):
                 filter_ledger_diffs(diffs)
             )
 
+    def _assert_no_migration_restrictions(self, domain_name):
+        assert should_use_sql_backend(domain_name)
+        assert not COUCH_SQL_MIGRATION_BLACKLIST.enabled(domain_name, NAMESPACE_DOMAIN)
+        assert not stale_get_export_count(domain_name)
+        assert not any(custom_report_domain == domain_name
+                       for custom_report_domain in settings.DOMAIN_MODULE_MAP.keys())
```

Comment:
could there be projects that use custom code that is not custom reports that would also need to be excluded?

Reply:
Yes, it's possible that there are others e.g. https://github.com/dimagi/commcare-hq/blob/014d03024d1aa792a84ed911bc385ca1f4ff60fa/custom/hki (though I think that one applies to the bihar domains which are already covered).

Reply:
@gcapalbo I'd believe that it's possible. What are the chances there is a nice, compiled list of those projects somewhere?

Would it be reasonable/sufficient to take the names from the list of dirs in `custom/`, `custom/_legacy`, `custom/apps`, `custom/reports`, and restrict those as well or do you have something else in mind?

Reply:
i don't think there's a list of projects anywhere, but i think looking at the directories you mentioned and seeing which ones don't already exist in the `DOMAIN_MODULE_MAP` would probably be the additional candidates to look out for. you may need to share the list with the dev team to find out which projects use them